api.yml

```
@@ -17,13 +17,38 @@ tags:
   - name: public
     description: Public APIs facing the Holder and Verifier.
 paths:
-  /credential:
+  /credentials/issueCredential:
     post:
       tags:
         - internal
       summary: Issues a credential and returns it in the response body.
       operationId: issueCredential
       description: Issues a credential and returns it in the response body. Support of this part of the API is REQUIRED for implementations.
+      requestBody:
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/IssueCredentialRequest'
+        description: Parameters for issuing the credential.
+      responses:
+        '201':
+          description: Credential successfully issued!
+          content:
+            application/json:
+              schema:
+                type: object
+                description: The issued credential. This SHOULD include a reference or ID that can be used later for other parts of the API, e.g. changing and retrieving the status of the credential, or refreshing the credential.
+        '400':
+          description: invalid input!
+        '500':
+          description: error!
+  /credentials/composeAndIssueCredential:
```

-1 to verbs in URLs, this URL has *two* verbs, *and* is creating a resource. Please read the following for how we should be naming URLs:

https://restfulapi.net/resource-naming/

There is an entire dissertation that's written about why this is a bad idea and breaks many of the design principles behind HTTP:

https://www.ics.uci.edu/~fielding/pubs/dissertation/rest_arch_style.htm

These URL paths violate some basic tenets of Web Architecture and RESTful design, which is okay to break from time to time and for good reason, but this isn't one of those times. We could easily avoid doing terrible things w/ the URLs here.

---

I agree, lets discuss here: https://github.com/w3c-ccg/vc-issuer-http-api/issues/41

---

Can I resolve this?

---

No discussion happened, can we make this:

```suggestion
  /credentials/compose
```

And presume that `compose` cannot possibly conflict with a reference ID? If we make this endpoint just compose a VC that can then be submitted via `/credentials` for issuance, then this would be a "controller" per the style guide as we're not creating a resource that is stored/referenced on the server. We also get composition and simpler APIs.

---

@peacekeeper -- If we have the `compose` API just compose and you turn around and submit the composed VC to `/credentials` would that work for you? If so, I could quickly agree to this and it would make implementation easier for a number of us, I would think. If not, could we at least call one endpoint: `/credentials` and then we can leave this one as is with the longer name.

---

We can handle these comments in another PR.